test/langtools/jdk/javadoc/lib/javadoc/tester/JavadocTester.java

```diff
@@ -975,6 +915,336 @@ private void diff(Path baseDir1, Path baseDir2, Path file) {
+         * @return this object
+         */
+        public OutputChecker setExpectFound(boolean expectFound) {
+            this.expectFound = expectFound;
+            return this;
+        }
+
+        /**
+         * Specifies whether matches are expected to be found in order or not.
+         * The default is {@code true}.
+         *
+         * @param expectOrdered  whether matches should be ordered
+         * @return this object
+         */
+        public OutputChecker setExpectOrdered(boolean expectOrdered) {
+            this.expectOrdered = expectOrdered;
+            return this;
+        }
+
+        /**
+         * Specifies whether matches are allowed to overlap.
+         * The default is {@code false}.
+         *
+         * @param allowOverlaps whether matches may overlap
+         * @return this object
+         */
+        public OutputChecker setAllowOverlaps(boolean allowOverlaps) {
+            this.allowOverlaps = allowOverlaps;
+            return this;
+        }
+
+        /**
+         * Checks for the presence (or absence) of a series of strings.
+         * Within the search strings, the newline character {@code \n}
+         * will be translated to the platform newline character sequence.
+         *
+         * @param strings the strings to be searched for
+         */
+        public OutputChecker check(String... strings) {
+            if (name == null) {
+                out.println("Skipping checks for:\n"
+                        + List.of(strings).stream()
+                        .map(s -> "    " + toShortString(s))
+                        .collect(Collectors.joining("\n")));
+                return this;
+            }
+
+            for (String stringToFind : strings) {
+                check(startPos -> findString(stringToFind, startPos), "text", stringToFind);
+            }
+            return this;
+        }
+
+        /**
+         * Checks for the presence (or absence) of a series of regular expressions.
+         * Unlike {@link #check(String...)}, there is no special handling for
+         * newline characters. Use {@code \R} to match the platform newline sequence.
+         *
+         * @param patterns the regular expressions to be searched for
+         */
+        public OutputChecker check(Pattern... patterns) {
+            if (name == null) {
+                out.println("Skipping checks for:\n"
+                        + List.of(patterns).stream()
+                        .map(p -> "    " + toShortString(p.pattern()))
+                        .collect(Collectors.joining("\n")));
+                return this;
+            }
+            for (Pattern pattern : patterns) {
+                check(startPos -> findPattern(pattern, startPos), "pattern", pattern.pattern());
+            }
+            return this;
+        }
+
+        /**
+         * Checks that there are no duplicate lines in the content.
+         */
+        public OutputChecker checkUnique() {
```

Comment:
checkUnique seems to be neither used nor tested. As far as I can see, it was like that before this PR. Consider removing this method.